ruby/ruby_assign.rb

```diff
@@ -0,0 +1,265 @@
+#input data from given assignment
+
+records = []
+
+record = {
+  student_id: 1,
+  department: "a1",
+  maths: 43,
+  physics: 54,
+  chemistry:65,
+  year: 2016,
+  count: 1
+}
+records << record
+
+record = {
+  student_id: 2,
+  department: "a1",
+  maths: 66,
+  physics: 52,
+  chemistry:65,
+  year: 2016,
+  count: 1
+}
+records << record
+
+record = {
+  student_id: 3,
+  department: "a7",
+  maths: 87,
+  physics: 32,
+  chemistry:43,
+  year: 2016,
+  count: 1
+}
+records << record
+
+record = {
+  student_id: 1,
+  department: "a1",
+  maths: 21,
+  physics: 52,
+  chemistry:65,
+  year: 2015,
+  count: 1
+}
+records << record
+
+record = {
+  student_id: 2,
+  department: "a1",
+  maths: 68,
+  physics: 50,
+  chemistry:65,
+  year: 2015,
+  count: 1
+}
+records << record
+
+record = {
+  student_id: 3,
+  department: "a7",
+  maths: 85,
+  physics: 22,
+  chemistry:43,
+  year: 2015,
+  count: 1
+}
+records << record
+
+record = {
+  student_id: 4,
+  department: "a7",
+  maths: 21,
+  physics: 22,
+  chemistry:13,
+  year: 2016,
+  count: 1
+}
+records << record
+
+#starting of the program
+
+#taking user input
+print "Group_by ?  "
+group_bi = gets.chomp
+group_bi.downcase!
+
+while (group_bi != "student_id" && group_bi != "department" && group_bi != "year")
```

Comment:
use until instead of while. handle cases where input is empty and non group_by element.